component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPUtils.java

```diff
@@ -0,0 +1,561 @@
+     * @return True if user account is locked.
+     * @throws AuthenticationFailedException Exception on authentication failure.
+     */
+    public static boolean isAccountLocked(AuthenticatedUser authenticatedUser) throws AuthenticationFailedException {
+
+        try {
+            return VoiceOTPServiceDataHolder.getInstance().getAccountLockService()
+                    .isAccountLocked(authenticatedUser.getUserName(), authenticatedUser.getTenantDomain(),
+                            authenticatedUser.getUserStoreDomain());
+        } catch (Exception e) {
+            String errorMessage = String.format("Error while validating account lock status of user: %s.",
+                    authenticatedUser.getUserName());
+            throw new AuthenticationFailedException(errorMessage, e);
+        }
+    }
+
+    /**
+     * Get Account Lock Connector Configs.
+     *
+     * @param tenantDomain Tenant domain.
+     * @return Account Lock Connector configs.
+     * @throws AuthenticationFailedException Exception on authentication failure.
+     */
+    public static Property[] getAccountLockConnectorConfigs(String tenantDomain) throws AuthenticationFailedException {
+
+        Property[] connectorConfigs;
+        try {
+            connectorConfigs = VoiceOTPServiceDataHolder.getInstance()
+                    .getIdentityGovernanceService()
+                    .getConfiguration(
+                            new String[]{
+                                    VoiceOTPConstants.PROPERTY_ACCOUNT_LOCK_ON_FAILURE,
+                                    VoiceOTPConstants.PROPERTY_ACCOUNT_LOCK_ON_FAILURE_MAX,
+                                    VoiceOTPConstants.PROPERTY_ACCOUNT_LOCK_TIME,
+                                    VoiceOTPConstants.PROPERTY_LOGIN_FAIL_TIMEOUT_RATIO
+                            }, tenantDomain);
+        } catch (Exception e) {
+            throw new AuthenticationFailedException("Error occurred while retrieving account lock connector " +
+                    "configuration", e);
+        }
+
+        return connectorConfigs;
+    }
+
+    /**
+     * Check whether the user being authenticated via a local authenticator or not.
+     *
+     * @param context Authentication context.
+     * @return Whether the user being authenticated via a local authenticator.
+     */
+    public static boolean isLocalUser(AuthenticationContext context) {
+
+        Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
+        if (stepConfigMap != null) {
+            for (StepConfig stepConfig : stepConfigMap.values()) {
+                if (stepConfig.getAuthenticatedUser() != null && stepConfig.isSubjectAttributeStep()) {
+                    if (stepConfig.getAuthenticatedIdP().equals(VoiceOTPConstants.LOCAL_AUTHENTICATOR)) {
+                        return true;
+                    }
+                    break;
+                }
+            }
+        }
+        return false;
```

Comment:
Let's consider using negations for the conditions to avoid unnecessary indentations.